agent/structs/structs.go

```diff
@@ -1140,6 +1140,82 @@ type IndexedNodeDump struct {
 	QueryMeta
 }
 
+// IndexedConfigEntries has its own encoding logic which differs from
+// ConfigEntryRequest as it has to send a slice of ConfigEntry.
+type IndexedConfigEntries struct {
+	Entries []ConfigEntry
+	QueryMeta
```

Comment:
Let's just add `Kind` here and make it only support single type in one response.